app/screens/sso/sso_with_redirect_url.tsx

```diff
@@ -0,0 +1,174 @@
+import urlParse from 'url-parse';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import {setDeepLinkURL} from '@actions/views/root';
+import FormattedText from '@components/formatted_text';
+import Loading from '@components/loading';
+import {Theme} from '@mm-redux/types/preferences';
+import Store from '@store/store';
+
+interface SSOWithRedirectURLProps {
+    customUrlScheme: string;
+    intl: typeof intlShape;
+    loginError: string;
+    loginUrl: string;
+    onCSRFToken: (token: string) => void;
+    onMMToken: (token: string) => void;
+    setLoginError: (value: string) => void;
+    theme: Theme
+}
+
+function SSOWithRedirectURL({
+    customUrlScheme,
+    intl,
+    loginError,
+    loginUrl,
+    onCSRFToken,
+    onMMToken,
+    setLoginError,
+    theme,
+}: SSOWithRedirectURLProps) {
+    const [error, setError] = React.useState<string>('');
+    const style = getStyleSheet(theme);
+
+    const redirectUrl = customUrlScheme + 'callback';
+
+    const init = (resetErrors?: boolean) => {
+        if (resetErrors !== false) {
+            setError('');
+            setLoginError('');
+        }
+        const parsedUrl = urlParse(loginUrl, true);
+        parsedUrl.set('query', {
+            ...parsedUrl.query,
+            redirect_to: redirectUrl,
+        });
+        const url = parsedUrl.toString();
+        try {
+            Linking.openURL(url);
+        } catch (e) {
+            setError(
+                intl.formatMessage({
+                    id: 'mobile.oauth.failed_to_open_link',
+                    defaultMessage: 'Failed to open the link',
+                }),
+            );
+        }
+    };
+
+    const onURLChange = ({url}: { url: string }) => {
+        if (url && url.startsWith(redirectUrl)) {
+            Store?.redux?.dispatch(setDeepLinkURL(''));
+            const parsedUrl = urlParse(url, true);
+            if (parsedUrl.query && parsedUrl.query.MMCSRF && parsedUrl.query.MMAUTHTOKEN) {
+                onCSRFToken(parsedUrl.query.MMCSRF);
+                onMMToken(parsedUrl.query.MMAUTHTOKEN);
+            } else {
+                setError(
+                    intl.formatMessage({
+                        id: 'mobile.oauth.failed_to_login',
+                        defaultMessage: 'Failed to login',
+                    }),
+                );
+            }
+        }
+    };
+
+    React.useEffect(() => {
+        Linking.addEventListener('url', onURLChange);
+        init(false);
+        return () => {
+            Linking.removeEventListener('url', onURLChange);
+        };
+    }, []);
+
+    return (
+        <SafeAreaView
+            style={style.container}
+            testID='sso.redirect_url'
+        >
+            {loginError || error ? (
+                <View style={style.errorContainer}>
+                    <View style={style.errorTextContainer}>
+                        <Text style={style.errorText}>{loginError || error}{'.'}</Text>
```

Comment:
Are we able to format this to something like this?
```suggestion
                        <Text style={style.errorText}>
                             {`${loginError || error}.`}
                         </Text>
```